src/main/java/org/mule/module/json/internal/JsonSchemaValidatorNetworkntWrapper.java

```diff
@@ -0,0 +1,107 @@
+import com.networknt.schema.SchemaValidatorsConfig;
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * This Wrapper make validations with library com.networknt
+ * Supports Json schema validations with version Draft V6, V7, 2019-09 and 2020-12
+ */
+public class JsonSchemaValidatorNetworkntWrapper extends JsonSchemaValidator {
+
+  private final JsonSchema jsonSchema;
+
+  public JsonSchemaValidatorNetworkntWrapper(ValidatorKey key, JsonNode jsonSchemaNode) {
+    super(key);
+    jsonSchema = loadSchemaLibrary(jsonSchemaNode, super.getSchemaLocation(), super.getSchemaRedirects());
+  }
+
+  @Override
+  public void validate(InputStream inputStream) {
+    JsonNode jsonNode = super.asJsonNode(inputStream);
+    Set<ValidationMessage> responseValidate;
+    try {
+      responseValidate = jsonSchema.validate(jsonNode);
+    } catch (Exception e) {
+
+      if (e.getMessage().contains("Reference") && e.getMessage().contains("cannot be resolved")) {
+        //TODO We must create a new error: INVALID_REFERENCE, to inform the user that the external references declared in the Schema cannot be accessed(W-12301483)
+        throw new MuleRuntimeException(createStaticMessage("Invalid Schema References"), e);
+      } else {
+        throw new MuleRuntimeException(createStaticMessage(
+                                                           "Exception was found while trying to validate against json schema. Content was: "
+                                                               + jsonNode.toString()),
+                                       e);
+      }
+    }
+
+    if (!responseValidate.isEmpty()) {
+      throw new SchemaValidationException("Json content is not compliant with schema: \n" + responseValidate,
+                                          responseValidate.toString());
+    }
+  }
+
+  /**
+   * Load Schema for com.networknt( Draft 6, 7, 2019-09 & 2020-12)
+   * If the schema comes from a TEXT, is loaded with a jsonNode
+   * but If the schema comes from a PATH, is loaded with this location
+   * because is needed to solve references (example: $ref to other schema file).
+   */
+  private com.networknt.schema.JsonSchema loadSchemaLibrary(JsonNode jsonNode, String schemaLocation,
+                                                            Map<String, String> schemaRedirects)
+      throws ModuleException {
+    try {
+      JsonSchemaFactory schemaFactory = JsonSchemaFactory.getInstance(SpecVersionDetector.detect(jsonNode));
+
+      if (schemaLocation == null) {
+        return schemaFactory.getSchema(jsonNode, getUriRedirectConfig(schemaRedirects));
+      } else {
+        return schemaFactory.getSchema(new URI(resolveLocationIfNecessary(schemaLocation)),
+                                       getUriRedirectConfig(schemaRedirects));
+      }
+    } catch (URISyntaxException e) {
+      throw new ModuleException(format("Could not load JSON schema [%s]. %s", schemaLocation, e.getMessage()),
```

Comment:
Please review the case: This exception message does not seem related to the one that can be thrown when using the JavaJsonTool lib. Can we make the cases more similar? They don't look like catching the same problems.